### PR TITLE
add match_mark

### PR DESCRIPTION
**Pull request by csschwe**
Add match_mark to be used to match set_mark in firewall rules.

### CODE REVIEW

File: lib/puppet/type/firewall.rb [hunk 1]
```diff
@@ -780,6 +780,33 @@ def insync?(is)
...
 
+  # match mark
+  newproperty(:match_mark, :required_features => :mark) do
+    desc <<-EOS
+      Match the Netfilter mark value associated with the packet.  Accepts either of:
+      mark/mask or mark.  These will be converted to hex if they are not already.
+    EOS
+    munge do |value|
+      int_or_hex = '[a-fA-F0-9x]'
+      match = value.to_s.match("(#{int_or_hex}+)(/)?(#{int_or_hex}+)?")
```
Comment (elyscape) [inline lib/puppet/type/firewall.rb]: A better matcher would be:
```ruby
mark_regex = %r{\A((?:0x)?[0-9A-F]+)(?:/((?:0x)?[0-9A-F]+))?\z}I
match = value.to_s.match(mark_regex)
```
This has the added benefit of requiring significantly less processing of the match object.

**Comment by jonnytpuppet** [comment]
Hi There

Before we can merge this, please add a couple of basic acceptance tests for this addition.

Thanks!

**Comment by jonnytpuppet** [comment]
Thank you for adding the iptables acceptance test. However, we still need an acceptance test for ip6table as well.

Thanks!

**Comment by csschwe** [comment]
Added ip6tabled acceptance test

File: spec/acceptance/match_mark_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            action     => reject,
+            provider => 'ip6tables',
+          }
+        EOS
+
+        apply_manifest(pp, :catch_failures => true)
+      end
+
+      it 'should contain the rule' do
+        shell('iptables-save') do |r|
```
Comment (jonnytpuppet) [inline spec/acceptance/match_mark_spec.rb]: If you are testing ip6tables, you need to run ip6tables-save to check the output.

Comment (csschwe) [inline spec/acceptance/match_mark_spec.rb]: Updated the ip6 part to use ip6tabes-save

**Comment by jonnytpuppet** [comment]
Closing due to #527 which bring in these changes as well as some other minor fixes.

**Pull request closed without merging**